types/node/v18/test/events.ts

```diff
@@ -130,3 +130,17 @@ async function test() {
     const eventEmitter = new events.EventEmitter();
     events.EventEmitter.setMaxListeners(42, eventTarget, eventEmitter);
 }
+
+{
+    class MyEmitter extends events.EventEmitterAsyncResource {}
+
+    const emitter = new MyEmitter({
+        name: "testing",
+        triggerAsyncId: 123,
+    });
+
+    emitter.asyncId // $ExpectType number
+    emitter.asyncResource // $ExpectType EventEmitterReferencingAsyncResource
+    emitter.triggerAsyncId // $ExpectType number
+    emitter.emitDestroy();
```

Comment:
Maybe adding also tests: 

- Direct EventEmitterAsyncResource instantiation 
- Child EventEmitterAsyncResource class instantiation without args

But it's probably a bit out of the scope of types definition code.  
I'm not familiar with the expected tests coverage in @types/node.